.github/workflows/ci.yml

```diff
@@ -0,0 +1,137 @@
+name: CI
+
+concurrency:
+  group: ci-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - staging
+      - trying
+  pull_request:
+    branches: [main]
+    paths:
+      - "**"
+      - "!/*.md"
+      - "!/**.md"
+
+jobs:
+  check:
+    name: Rust Check - ${{ matrix.check }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        rust: [stable]
+        rust-target: [x86_64-unknown-linux-gnu]
+        check: [test, check, fmt, clippy]
+    env:
+      RUST_BACKTRACE: full
+      RUSTV: ${{ matrix.rust }}
+      TARGET: ${{ matrix.rust-target }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Rust stable
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          components: clippy
+      - uses: Swatinem/rust-cache@v2
+        with:
+          key: ${{ matrix.rust-target }}-${{ matrix.check }}
+      - name: Cargo test
+        if: ${{ matrix.check == 'test' }}
+        run: |
+          cargo test
+      - name: Cargo check
+        if: ${{ matrix.check == 'check' }}
+        run: |
+          cargo check --all-targets
+
+      - name: Cargo fmt
+        if: ${{ matrix.check == 'fmt' }}
+        run: cargo fmt -- --check
+
+      - name: Cargo clippy
+        if: ${{ matrix.check == 'clippy' }}
+        run: cargo clippy --all-targets -- -D warnings
+
+  integration_tests:
```

Comment:
Same comment as for makefile. No need to add CI for non-existing tests.

Reply:
removed